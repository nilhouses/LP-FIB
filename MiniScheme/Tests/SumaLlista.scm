;Aquest test es pot executar des del directori arrel del projecte amb les comandes:

  ;SI VOLEM COMPROVAR EL CORRECTE FUNCIONAMENT DE LA GRAMÃ€TICA
    ;python3 scheme.py ./Tests/SumaLlista.scm | diff - ./Tests/SumaLlista.out

  ;SI VOLEM VEURE LA SORTIDA PER PANTALLA
    ;python3 scheme.py ./Tests/SumaLlista.scm

;Consisteix en mostrar per pantalla la suma dels nombres de la llista l

(define l '(1 2 3 4 5 6 7 8 9 10))

(define (suma-llista llista)
  (if (null? llista)
      0
      (+ (car llista) (suma-llista (cdr llista)))))


(define (main)
  (display "l: ")
  (display l)
  (newline)

  (display "Suma dels elements d'l: ")
  (display (suma-llista l))
  (newline)
)

