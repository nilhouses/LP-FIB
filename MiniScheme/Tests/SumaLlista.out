l: ( 1 2 3 4 5 6 7 8 9 10 )
Suma dels elements d'l: 55
